torch/ao/quantization/utils.py

```diff
@@ -547,6 +547,89 @@ def _patched_module_call(self, *args, **kwargs):
     return fqn_to_example_inputs
 
 
+def _get_fixed_params_observed_lstm(
```

Comment:
can we provide a module the user can use directly instead of providing a util function?  Or, can we just make the quantizeable module do this by default?

Reply:
behavior which would make sense to me from user's perspective:
* user is quantizing for backend X (default: fbgemm/qnnpack)
* user is using a get_default_qconfig_mapping(backend_str) function
* for static quant, the right qconfig needs to be passed to LSTM
* if LSTM is using custom modules, we need to follow the custom module spec in how things are assigned qconfigs and/or quantized.  Does it even make sense at all to "assign parent qconfig to all child modules" for custom modules?

we should really understand the custom module spec and role in quantization (where is that defined)?

option 1:
* LSTM keeps using custom modules, we assign qconfig according to custom module spec (parent framework never assigns qconfig of insides of custom module, that is the user's job)

option 2:
* LSTM does not use custom modules, and then it can use qconfig_mapping

Reply:
option 1 right now I think, we are delaying the design of custom module in reference quantized model flow due to PT2 export, I think we can come up with the design in PT2 export

Reply:
As discussed offline, the problem with the current LSTM flow (before this PR) is that the user can specify the QConfig for `torch.nn.LSTM` through QConfigMapping, but this breaks the custom module abstraction where the custom module itself is responsible for inserting its own observers. The new flow enabled in this PR respects this abstraction.

To fix the default behavior, we can check QConfigMapping and throw an exception if we find `torch.nn.LSTM` (and MHA) with a helpful message to redirect users towards the right behavior, which is to manually set (`model.lstm.qconfig = QConfig(...)`), which respects the custom module abstractions. Since this is technically a separate issue, it will be fixed in a future PR.